### PR TITLE
#O21KB-13-Front-End: Add new warehouse Component (Responsive)

### PR DESCRIPTION
**Pull request by nilouhou**
[Jira Ticket #O21KB-13](https://brainstationeducation.atlassian.net/browse/O21KB-13)

I used @manecipoglu form component, could not make it  as elegant as yours with FormSection though with error messaging, So maybe later after demo we could refactor it better. Right now, it has validation functionality and on and off with error message

### CODE REVIEW

File: client/src/pages/AddWarehouse/AddWarehouse.jsx [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+								</label>
+								<input
+									onChange={this.handleChange}
+									type="text"
+									id="streetAddress"
+									name="streetAddress"
+									className="form-field__input"
+								/>
+								{this.state.error && this.state.error !== undefined ? (
+									<div className="error">
```
Comment (shaynelar) [inline client/src/pages/AddWarehouse/AddWarehouse.jsx:140]: if this.state.error is truthy, it wont ever be undefined. So I'm not sure you need to check if it isn't undefined

File: client/src/pages/AddWarehouse/AddWarehouse.jsx [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+			contactName: "",
+			position: "",
+			phoneNumber: "",
+			email: "",
+			error: "",
+		};
+
+		this.handleChange = this.handleChange.bind(this);
+		this.handleSubmit = this.handleSubmit.bind(this);
+	}
```
[the remark this answers is not in the document]
Comment (manecipoglu) [reply client/src/pages/AddWarehouse/AddWarehouse.jsx:26]: In addition to what Shayne said, if you use the short notation for class components (state = {}) you don't need to type constructor. Even though React will complete that part for you at compilation, not writing it mean less code and less code is easier to maintain.

File: client/src/pages/AddWarehouse/AddWarehouse.jsx [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+
+	async componentDidMount() {
+		const warehouse = await axios(
```
Comment (manecipoglu) [inline client/src/pages/AddWarehouse/AddWarehouse.jsx:29]: You can wrap this in a try - catch block. I forgot to do mine as well :)

File: client/src/pages/AddWarehouse/AddWarehouse.jsx [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+	};
+
+	handleSubmit(event) {
+		event.preventDefault();
+
+		if (this.isFormValid()) {
+			const newWarehouse = this.state;
+			axios
+				.post(
+					`${API_URL}/warehouses/${this.props.match.params.id}`,
```
Comment (manecipoglu) [inline client/src/pages/AddWarehouse/AddWarehouse.jsx:87]: Another opportunity for a try-catch

**Comment by manecipoglu** [comment]
This is another big development for our project. Thank you Nilou! I haven't had the chance to try it on my local machine but I will definitely do it tomorrow. I'm going to wait for you to resolve merge conflicts before merging.

**Comment by manecipoglu** [comment]
I decided to merge this to have a better presentation. But let's resolve the issues anyway. Thank you!

File: client/src/pages/AddWarehouse/AddWarehouse.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (nilouhou) [reply client/src/pages/AddWarehouse/AddWarehouse.jsx]: Thanks Shane, I understand you point. I did that at the beginning with out whole constructor and binding but the validation function was not triggered and got errors, but after I bind it, it works so I decide to keep it :)

File: client/src/pages/AddWarehouse/AddWarehouse.jsx (hunk 1 above)
Comment (nilouhou) [reply client/src/pages/AddWarehouse/AddWarehouse.jsx]: sure, will fix it

**Comment by nilouhou** [comment]
@manecipoglu thank you for your comments, I noted, will apply them, and thank you for resolving the conflict and staying up that late.

**Pull request merged**